### DISCUSSION TITLE
[Chapter 11] 스프링 스케줄링 테스크

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/spring-boot-for-msa#186] [Chapter 11. Spring Scheduling Tasks] [answered]
Chapter 11 들어가면서 아래와 설명이 나온다.
 
> 배치 프로세스 중에서도 처리하는 데이터 크기가 비교적 작고, 처리 주기가 비교적 짧다면 이를 마이크로 배치라고 한다.   
  ... 
스프링 프로젝트에서는 배치 프로세스를 위한 여러 가지 기능과 프레임워크를 제공한다. 스프링 배치 프레임워크는 다수의 데이터를 체계적으로 처리하는 기능을 제공하는데, `여기에서는 그중 마이크로 배치를 위한 방법에 집중한다. 특히 마이크로 배치를 실행할 수 있는 스케줄링 기능과 트리거를 설명한다.`
 

Q1. 그럼 마이크로 배치에 상반되는 데이터 크기가 매우 크고 처리 주기가 짧지 않은 배치도 있을 것같다. 
그런 배치 프로세싱을 처리할 상황이 뭐가 있을까?  
그리고 그런 배치 프로세싱은 책에서 설명하는 배치 서버 아키텍처나 스케줄링 기능 등 설정 방법 등이 어떻게 다를까?

 

Q2. 회사서  `Spring Quartz` 라이브러리를 통해 배치 프로세싱을 구현하고 있다.  아직 코드 분석을 못해서 자세한 내용을 모르겠지만..
Spring Quartz 사용한 이유가 따로 있을까?

**Comment by corock** [comment, marked as answer]
> Q1. 그럼 마이크로 배치에 상반되는 데이터 크기가 매우 크고 처리 주기가 짧지 않은 배치도 있을 것같다.
그런 배치 프로세싱을 처리할 상황이 뭐가 있을까?
그리고 그런 배치 프로세싱은 책에서 설명하는 배치 서버 아키텍처나 스케줄링 기능 등 설정 방법 등이 어떻게 다를까?

리테일 서비스를 예로 들어보면, "매일 오후 4시에 상품 주문 배송 정보를 고객들에게 문자로 일괄 전송한다." 라는 정책이 있다고 해봅시다.
확보한 고객이 많은 서비스의 경우 문자를 보낼 대상이 많기 때문에 데이터 크기에 영향을 주고, 배치 프로세싱에도 영향도가 이어질 것 같습니다.
이와 관련한 [if(kakao)dev2022 컨퍼런스](https://www.youtube.com/watch?v=2IIwQDIi3ys) 세션이 있어 같이 첨부합니다.

> Q2. 회사서 Spring Quartz 라이브러리를 통해 배치 프로세싱을 구현하고 있다. 아직 코드 분석을 못해서 자세한 내용을 모르겠지만..
Spring Quartz 사용한 이유가 따로 있을까?

Spring Quartz 는 배치 프로세싱을 직접 관장한다기보다 스케줄링을 도와주는 라이브러리로 알고 있습니다.
정해진 스케줄링에 따라 Spring Quartz 를 통해 Spring Batch 를 호출해서 배치 프로세싱을 하는 구조로 사용하고 있지 않나요?
혹시 사용 경험이 있으신 분이 있다면 답글 부탁드립니다~
